pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -348,24 +348,27 @@ func generateOriginalKubeletConfigWithFeatureGates(cc *mcfgv1.ControllerConfig,
 	if err != nil {
 		return nil, fmt.Errorf("could not deserialize the Kubelet source: %v", err)
 	}
-
-	featureGates, err := generateFeatureMap(features, openshiftOnlyFeatureGates...)
-	if err != nil {
-		return nil, fmt.Errorf("could not generate features map: %v", err)
-	}
-
-	// Merge in Feature Gates.
-	// If they are the same, this will be a no-op
-	if err := mergo.Merge(&originalKubeConfig.FeatureGates, featureGates, mergo.WithOverride); err != nil {
-		return nil, fmt.Errorf("could not merge feature gates: %v", err)
+	if features != nil {
+		featureGates, err := generateFeatureMap(features, openshiftOnlyFeatureGates...)
+		if err != nil {
+			return nil, fmt.Errorf("could not generate features map: %v", err)
+		}
+		// Merge in Feature Gates.
+		// If they are the same, this will be a no-op
+		if err := mergo.Merge(&originalKubeConfig.FeatureGates, featureGates, mergo.WithOverride); err != nil {
+			return nil, fmt.Errorf("could not merge feature gates: %v", err)
+		}
```

Comment:
I think there are two options for solving the nil `features` here, the approach you have here, which then relies on the template feature gates being correct/synced. If we go with this approach I'd suggest a comment such as
```
// When no feature gate is provided, fall back to default features in the kubelet config template
```

The alternative is to allow `generateFeatureMap` to handle a nil `features` by returning the default feature map when passed a nil. This would mean that `generateFeatureMap` is always the source of the features and inconsistency between the code and the templates would no longer matter, wondering if this might be preferable and less bug prone in the future, WDYT?

Reply:
Thanks! Soloving the nil features in `generateFeatureMap` sounds good to me. I add the code to generate default feature map from the createNewDefaultFeatureGate(). PTAL. https://github.com/openshift/machine-config-operator/pull/2547/files#diff-bbd76e72ae643598c17f41b62eb05922036ddc6275420eded1c9bc2310a06215R178